docs/src/test/scala/docs/ApiMayChangeDocCheckerSpec.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+package docs
+
+import java.lang.reflect.Method
+
+import akka.annotation.ApiMayChange
+import org.reflections.Reflections
+import org.reflections.scanners.{ MethodAnnotationsScanner, TypeAnnotationsScanner }
+import org.reflections.util.{ ClasspathHelper, ConfigurationBuilder }
+import org.scalatest.{ Matchers, WordSpec }
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable
+import scala.io.Source
+
+class ApiMayChangeDocCheckerSpec extends WordSpec with Matchers {
+
+  def prettifyName(clazz: Class[_]): String = {
+    clazz.getCanonicalName.replaceAll("\\$minus", "-").split("\\$")(0)
+  }
+
+  // As Specs, Directives and HttpApp inherit get all directives methods, we skip those as they are not really bringing any extra info
+  def removeClassesToIgnore(method: Method): Boolean = {
+    Seq("Spec", ".Directives", ".HttpApp").exists(method.getDeclaringClass.getCanonicalName.contains)
```

Comment:
I guess this can be done with some custom scanner on the `ConfigurationBuilder` but I didn't find it in the documentation